lib/badge-data.spec.js

```diff
@@ -84,4 +85,11 @@ describe('Badge data helpers', function() {
       colorB: '#f00bae',
     });
   });
+
+  test(setBadgeColor, () => {
+    given({}, 'red').expect({ colorscheme: 'red' });
+    given({}, 'f00f00').expect({ colorB: '#f00f00' });
+    given({ colorB: '#f00f00', colorscheme: 'blue' }, 'red').expect({ colorB: '#f00f00', colorscheme: 'red' });
+    given({ colorB: '#b00b00', colorscheme: 'blue' }, 'f00f00').expect({ colorB: '#f00f00', colorscheme: 'blue' });
```

Comment:
```js
given({ colorB: '#f00f00', colorscheme: 'blue' }, 'red').expect({ colorB: '#f00f00', colorscheme: 'red' });
given({ colorB: '#b00b00', colorscheme: 'blue' }, 'f00f00').expect({ colorB: '#f00f00', colorscheme: 'blue' });
```

These outcomes would confuse me as a caller. Could we implement a clearer behavior?

Reply:
@paulmelnikow Is this behavior clearer for you?
```js
given({ colorB: '#f00f00', colorscheme: 'blue' }, 'red').expect({ colorscheme: 'red' });
given({ colorB: '#b00b00', colorscheme: 'blue' }, 'f00f00').expect({ colorB: '#f00f00' });
```
Do we want to handle `colorA` in `setBadgeColor`?

Reply:
Yea, that behavior seems much clearer!

Re `colorA`, yea, we probably should! Do you want to address that before we land this?

Reply:
I could handle `colorA` as well, but I'm not sure how to do this and is it worth. `colorB` is set 34 times in server.js (in my branch), `colorA` - once. 
If we want to handle colorA, colorB and colorscheme in `setBadgeColor`, `setBadgeColor` could take `color` object as a parameter, where this object could have properties `colorA`, `colorB` and `colorscheme`. 
On the other hand spread syntax can handle this too (but it will not clear remaining properties unless we set null for them):
```
let badgeData = {text: 'a', colorscheme: 'red'}
badgeData = { ...badgeData, colorB: '#001122', colorscheme: null }
```

Reply:
It seems like this is still pending. I'm fine with ignoring `colorA` for the moment, handling only `colorB` and `colorscheme`.